test/testdata/resolver/enum_val_types_order1__file1.rb

```diff
@@ -0,0 +1,15 @@
+# typed: true
```

Comment:
Our test suite has no coverage of "generic T::Enum"

There was a point in the past when T::Enum was a generic (it included Enumerable) and it's not unreasonable to think that it might become a generic in the future (you'd be required to specify what type the enum serializes to, so that `MyEnum::X.serialize` could return `String` or `Integer` or `Symbol` instead of `T.untyped`)

"generics + T::Enum" could presumably have codepaths that are only triggered by paths introduced in this PR. would you mind adding some tests as such? [for example, here's a starter](https://sorbet.run/#%23%20typed%3A%20true%0Aclass%20MyEnum%20%3C%20T%3A%3AEnum%0A%20%20extend%20T%3A%3AGeneric%0A%20%20Value%20%3D%20type_template(fixed%3A%20String)%0A%0A%20%20enums%20do%0A%20%20%20%20X%20%3D%20new%0A%20%20%20%20Y%20%3D%20new%0A%20%20%20%20Z%20%3D%20new%0A%20%20end%0Aend%0A%0Aextend%20T%3A%3ASig%0Asig%20%7Bparams(xy%3A%20T.any(MyEnum%3A%3AX%5BInteger%5D%2C%20MyEnum%3A%3AY%5BInteger%5D)).void%7D%20%23%20ok%0Adef%20foo(xy)%0Aend%0A%0AXY%20%3D%20T.type_alias%20%7BT.any(MyEnum%3A%3AX%5BInteger%5D%2C%20MyEnum%3A%3AY%5BInteger%5D)%7D%20%23%20error)

such tests will likely have lots of errors, and it's fine to say that we expect them, but it should make sure we don't have for example: a nullptr that we were expecting to be non-null.